lib/galaxy/tools/deps/conda_util.py

```diff
@@ -266,6 +269,42 @@ def hash_conda_packages(conda_packages, conda_target=None):
     return h.hexdigest()
 
 
+def is_conda_installed(conda_context):
+    """
+    Check if conda_exec exists
+    """
+    if os.path.exists(conda_context.conda_exec):
+        return True
+    else:
+        return False
+
+
+def can_install_conda(conda_context):
+    """
+    If conda_exec is set to a path outside of conda_prefix,
+    there is no use installing conda into conda_prefix, since it can't be used by galaxy.
+    If conda_exec equals conda_prefix/bin/conda, we can install conda if either conda_prefix
+    does not exist or is empty.
+    """
+    conda_exec = os.path.abspath(conda_context.conda_exec)
+    conda_prefix_plus_exec = os.path.abspath(os.path.join(conda_context.conda_prefix, 'bin/conda'))
+    if conda_exec == conda_prefix_plus_exec:
+        if not os.path.exists(conda_context.conda_prefix):
+            return True
+        elif os.listdir(conda_context.conda_prefix) == []:
+            os.rmdir(conda_context.conda_prefix)  # Conda's install script fails if path exists (even if empty).
+            return True
+        else:
+            log.warning("Cannot install Conda because conda_prefix '%s' exists and is not empty.",
+                        conda_context.conda_prefix)
+            return False
+    else:
+        log.warning("Skipping installation of Conda into conda_prefix '%s', "
+                    "since conda_exec '%s' is set to a path outside of conda_prefix.",
+                    conda_context.conda_prefix, conda_context.conda_exec)
+        return False
```

Comment:
That makes sense (and we can use this to provide admins more info about a resolvers status)